app/src/main/java/com/android/sooz/filterapp/models/StarWarsSpecies.java

```diff
@@ -0,0 +1,38 @@
+package com.android.sooz.filterapp.models;
+
+import android.content.Intent;
+
+//class that models the species data for use in the list
+public class StarWarsSpecies {
+    public String name;
+    public String classification;
+    public String language;
+    public String homeworld;
+
+    //creates property values for construction a species in the application
+    public StarWarsSpecies(String name, String classification, String language, String homeworld){
+        this.name = name;
+        this.classification = classification;
+        this.language = language;
+        this.homeworld = homeworld;
+    }
+
+    //allows desired species data object to be passed from to an activity/view
+    public void fillIntent(Intent intent){
```

Comment:
This is a perfectly reasonable approach in this case since the data is small. The other approach that is often taken, especially when your model starts to get bigger, is to pass some sort of id in the intent rather than all the data, and have the receiver just look up the correct record. No need to change this, just something to keep in mind.